tokio/src/io/mod.rs

```diff
@@ -265,6 +265,8 @@ cfg_io_std! {
 cfg_io_util! {
     mod split;
     pub use split::{split, ReadHalf, WriteHalf};
+    mod join;
+    pub use join::Join;
```

Comment:
Similar to `split`, I would also introduce a top-level function for constructing this type.